packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -609,25 +610,26 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		assert(revertibleBranch !== undefined, 0x7cc /* expected to find a revertible commit */);
 		const commitToRevert = revertibleBranch.getHead();
 
-		let change = this.changeFamily.rebaser.invert(
-			tagChange(commitToRevert.change, revision),
-			false,
+		let change = makeAnonChange(
+			this.changeFamily.rebaser.invert(tagChange(commitToRevert.change, revision), false),
 		);
```

Comment:
Why not tag the change from the start?
```suggestion
		let change = tagChange(
			this.changeFamily.rebaser.invert(tagChange(commitToRevert.change, revision), false),
			this.mintRevisionTag(),
		);
```

Reply:
I agree that this is a bit odd. Currently `branch.apply` on line 629 does the revision tag inlining, so it seems like we shouldn't do inlining here. But if we don't do inlining here then the changeset should be considered as having an undefined revision. Maybe it would make more sense for `branch.apply` to take a tagged change which is expected to already have revisions inline.

Reply:
I think it would. I'm okay with doing that in a separate PR if you prefer.